packages/ui/src/Card.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FC } from 'react'
+import { Icon, IconProps } from './Icon'
+import { DataTestProp } from '@hazelcast/helpers'
+
+import styles from './Card.module.scss'
+
+export type CardProps = {
+  title: string
+  icon?: IconProps['icon']
+  separator?: boolean
+} & DataTestProp
+
+/**
+ * ### Purpose
+ * Card components define content hierarchy (visual and logical), separating content into "groups".
+ *
+ * ### General Info
+ * - Card can be rendered with an optional separator and/or icon.
+ */
+export const Card: FC<CardProps> = ({ title, icon, separator = false, 'data-test': dataTest, children }) => (
+  <div data-test={dataTest ?? 'card-wrapper'} className={styles.wrapper}>
+    {title && (
+      <>
+        <h3 data-test="card-heading" className={styles.heading}>
+          {icon && <Icon icon={icon} className={styles.icon} ariaHidden />}
+          {title}
+        </h3>
+      </>
+    )}
+
+    {separator && <span data-test="card-separator" className={styles.separator} />}
```

Comment:
In `Card.module.scss` we're using `display: block;` - can't we just change this element to `div` and remove the `display: block;`?

Reply:
Well, I think we can, but will it improve anything? It's just a habit of mine to use `span` elements as separators.

Reply:
@danad02 Maybe I could replace it with `<hr />`?

Reply:
@jozefhruska I take it back, `<hr />` should be used to represent a break between paragraphs.  

From https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr:
> Historically, this has been presented as a horizontal rule or line. While it may still be displayed as a horizontal rule in visual browsers, this element is now defined in semantic terms, rather than presentational terms, so if you wish to draw a horizontal line, you should do so using appropriate CSS.

I think we should use the `div` with styles as we have now.

Reply:
nit: I feel like we could not add new elements to the DOM and simply use `border-bottom` on the `h3`. WDYT?

Reply:
> Well, I think we can, but will it improve something? It's just a habit of mine to use span elements as separators.

It won't improve anyting visually or semantically. It's just that changing span's display to block is an extra step during implementation - div is a block element by default.

Reply:
@danad02 Replaced with `<div>` 👍